### PR TITLE
session,executor: add variables to support set if enable tiflash pipeline model by sql statement

### LINKED ISSUE #44468: Add a way to set if enable tiflash pipeline model by sql statement
## Enhancement

After https://github.com/pingcap/tiflash/pull/7597, tiflash can enable the pipeline model without restarting.
However, the config can only be changed by TiFlash's configure file, which is not easy to use, we'd better add a way to set  if enable tiflash pipeline model by sql statement.

### PR DESCRIPTION
**Pull request by SeaRise**
### What problem does this PR solve?

Issue Number: close #44468

Problem Summary:

### Check List

Tests 

- [ ] Unit test
- [ ] Integration test
- [ ] Manual test (add detailed scripts or steps below)
- [ ] No code

Side effects

- [ ] Performance regression: Consumes more CPU
- [ ] Performance regression: Consumes more Memory
- [ ] Breaking backward compatibility

Documentation

- [ ] Affects user behaviors
- [ ] Contains syntax changes
- [ ] Contains variable changes
- [ ] Contains experimental features
- [ ] Changes MySQL compatibility

### Release note

Please refer to [Release Notes Language Style Guide](https://pingcap.github.io/tidb-dev-guide/contribute-to-tidb/release-notes-style-guide.html) to write a quality release note.

```release-note
None
```

### CODE REVIEW

**Review by windtalker — APPROVED** [review]
LGTM

File: sessionctx/variable/session.go [hunk 1]
```diff
@@ -872,6 +872,12 @@ type SessionVars struct {
 	// not limit and spill will never happen
 	TiFlashMaxBytesBeforeExternalSort int64
 
+	// TiFlashEnablePipelineMode means if we should enable tiflash pipeline model to execute query or not.
```
Comment (XuHuaiyu) [inline sessionctx/variable/session.go:875]: Will `TiFlashExecutionModel` be better?

File: sessionctx/variable/session.go [hunk 2]
```diff
@@ -872,6 +872,12 @@ type SessionVars struct {
...
+	// Default value is -1, means that whether to execute the query with pipeline model to is determined by the configuration of the tiflash node.
```
[the remark this answers is not in the document]
Comment (SeaRise) [reply sessionctx/variable/session.go]: After discussion, it is now changed to `bool tidb_enable_tiflash_pipeline_model{true, false}`

File: sessionctx/variable/session.go (hunk 1 above)
Comment (SeaRise) [reply sessionctx/variable/session.go]: After discussion, it is now changed to `bool tidb_enable_tiflash_pipeline_model{true, false}`

**Review by XuHuaiyu — APPROVED** [review]

**Pull request merged**